sql/src/main/java/io/crate/analyze/expressions/ExpressionAnalysisContext.java

```diff
@@ -24,16 +24,40 @@
 import io.crate.analyze.symbol.Function;
 import io.crate.analyze.symbol.Symbol;
 import io.crate.metadata.FunctionInfo;
+import io.crate.sql.tree.ArrayComparisonExpression;
+import io.crate.sql.tree.Expression;
 
+import java.util.IdentityHashMap;
 import java.util.List;
+import java.util.Map;
 
 public class ExpressionAnalysisContext {
 
+    private final Map<Expression, Object> arrayExpressions = new IdentityHashMap<>();
+
     public boolean hasAggregates = false;
 
     Function allocateFunction(FunctionInfo functionInfo, List<Symbol> arguments) {
         Function newFunction = new Function(functionInfo, arguments);
         hasAggregates = hasAggregates || functionInfo.type() == FunctionInfo.Type.AGGREGATE;
         return newFunction;
     }
+
+    /**
+     * Registers the given as an array comparison expression. Can be used by downstream
+     * operators to check if an expression is part of an {@link ArrayComparisonExpression}.
+     * @param arrayExpression the expression to register
+     */
+    void registerArrayExpression(Expression arrayExpression) {
```

Comment:
Oh I miss-read that - given the name `registerArrayExpression` I assumed that it's registering the `ArrayComparisonExpression`.

Maybe rename this to `registerArrayChild` then? (Or switch to a `insideArrayExpression` context-flag pattern.)